link-grammar/dict-common/dict-utils.c

```diff
@@ -297,7 +297,7 @@ static bool exp_has_connector(const Exp * e, int depth,
  **/
 bool is_exp_like_empty_word(Dictionary dict, Exp *exp)
 {
-	const char *cs = string_set_lookup(EMPTY_CONNECTOR, dict->string_set);
+	const char *cs = linkgrammar_get_dict_define(dict, EMPTY_CONNECTOR);
```

Comment:
It may not be a good idea to use `linkgrammar_get_dict_define()` in parse run time due to these reasons:
1. It is currently a "costly" operation, as it linearly scans a list of definitions. (In any case, I should re-implement it as a dict word starting with `\4` and has a string value instead of an expression.)
2. In general connectors are expected to use the dict string set, and current /future code may depend on that for connector comparison. E.g. `exp_has_connector()`, which is invoked by `is_exp_like_empty_word()`, uses `string_set_cmp(e->condesc->string, cs)`.

Solutions:
1. Add `Dictionary::empty_connector`. Invoke `linkgrammar_get_dict_define(dict, EMPTY_CONNECTOR)` once, `dictionary_setup_defines()` to set it.
2. Use here and elsewhere `string_set_lookup(empty_connector, dict->string_set)` to make sure it is from the dict string set.

Reply:
OK, I pushed an update for solution 1.